tests/prefetch_related/test_prefetch_related_objects.py

```diff
@@ -32,6 +34,34 @@ def setUpTestData(cls):
         cls.reader1.books_read.add(cls.book1, cls.book4)
         cls.reader2.books_read.add(cls.book2, cls.book4)
 
+    def test_add_clears_prefetched_objects(self):
+        p = Publisher.objects.get(pk=self.publisher.pk)
+        new_book = Book.objects.create(title='Ulysses')
+        prefetch_related_objects([p], 'book_set')
+        self.assertEqual(
+            set(p.book_set.all()),
```

Comment:
Is the set() stuff to avoid ordering differences? Usually we use `assertCountEqual` for that. (I guess you copied existing tests.)